testsuite/MDAnalysisTests/lib/test_pkdtree.py

```diff
@@ -0,0 +1,111 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- http://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+from __future__ import print_function, absolute_import
+from six.moves import zip
+
+import numpy as np
+from numpy.testing import TestCase, assert_array_equal, assert_equal
+
+from MDAnalysis.lib.pkdtree import PeriodicKDTree
+
+
+class TestPeriodicKDTree(TestCase):
```

Comment:
we use [pytest](https://www.pytest.org) for writing tests. That means no `TestCase` and no `setUp` methods. In fact you don't need a class for these tests. They can be individual functions. The only parameter you *reuse* is `self.box` but you can also replace it with a generic `np.ones(3)` in `test_set_coords`. For the last two tests a fixture would be best. Below is an example how to do that.

```python
@pytest.fixture
def ptree():
    box = np.ones(3) * 10
    coords = np.array([[2, 2, 2],
                                [5, 5, 5],
                                [1.1, 1.1, 1.1],
                                [11, -11, 11],  # wrapped to [1, 9, 1]
                                [21, 21, 3]],  # wrapped to [1, 1, 3]
                               dtype=np.float32)
    t = PeriodicKDTree(box)
    t.set_coords(coords)
    return t

def test_centers(tree):
    # do setup stuff for queries and centers
    for q, cs in zip(queries, centers):
        q = np.array(q, dtype=np.float32)
        cs = [np.array(c, dtype=np.float32) for c in cs]
        assert_equal(tree.find_centers(q, radius), cs)
```

Reply:
will do :)